protocol-units/execution/opt-executor/src/executor/transaction_pipe.rs

```diff
@@ -32,78 +32,76 @@ impl Executor {
 	) -> Result<(), TransactionPipeError> {
-		let mut mempool_client_receiver = self.mempool_client_receiver.write().await;
-		for _ in 0..256 {
-			// use select to safely timeout a request for a transaction without risking dropping the transaction
-			// !warn: this may still be unsafe
-			tokio::select! {
-				_ = tokio::time::sleep(tokio::time::Duration::from_millis(5)) => { () },
-				request = mempool_client_receiver.next() => {
-					match request {
-						Some(request) => {
-							match request {
-								MempoolClientRequest::SubmitTransaction(transaction, callback) => {
-									// Pre-execute Tx to validate its content.
-									// Re-create the validator for each Tx because it uses a frozen version of the ledger.
-									let vm_validator = VMValidator::new(Arc::clone(&self.db.reader));
-									let tx_result = vm_validator.validate_transaction(transaction.clone())?;
-
-									let status = if let Some(vm_status) = tx_result.status() {
-										// If the verification failed, return the error status.
-										let ms = MempoolStatus::new(MempoolStatusCode::VmError);
-										(ms, Some(vm_status))
-									} else {
-										// add to the mempool
-										let mut core_mempool = self.core_mempool.write().await;
-
-										debug!("Adding transaction to mempool: {:?} {:?}", transaction, transaction.sequence_number());
-										let status = core_mempool.add_txn(
-											transaction.clone(),
-											0,
-											transaction.sequence_number(),
-											TimelineState::NonQualified,
-											true
-										);
-
-										match status.code {
-											MempoolStatusCode::Accepted => {
-												debug!("Transaction accepted: {:?}", transaction);
-											},
-											_ => {
-												debug!("Transaction not accepted: {:?}", status);
-												Err(TransactionPipeError::TransactionNotAccepted(status))?;
-											}
-										}
-
-										// send along to the receiver
-										transaction_channel.send(transaction).await.map_err(
-											|e| anyhow::anyhow!("Error sending transaction: {:?}", e)
-										)?;
-
-										// report status
-										let ms = MempoolStatus::new(MempoolStatusCode::Accepted);
-										(ms, None)
-									};
-
-									if callback.send(Ok(status)).is_err() {
-										debug!("submit_transaction request has been canceled");
-									}
-								},
-								MempoolClientRequest::GetTransactionByHash(hash, sender) => {
-									let mempool = self.core_mempool.read().await;
-									let mempool_result = mempool.get_by_hash(hash);
-									if sender.send(mempool_result).is_err() {
-										debug!("get_transaction_by_hash request has been canceled");
-									}
+		// Drop the receiver RwLock as soon as possible.
+		let next = {
+			let mut mempool_client_receiver = self.mempool_client_receiver.write().await;
+			mempool_client_receiver.next().await
+		};
+		
+		if let Some(request) = next  {
```

Comment:
When the stream is closed, assuming it's fused, `next` will be `None` and the `tick_transaction_pipe` call with return `Ok(())` indefinitely. Perhaps it's better to return an error in this case like in #244.

Reply:
Done in [9e10091](https://github.com/movementlabsxyz/movement/pull/243/commits/9e10091af6867347029b877c3582f2037e52bdb9).